pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,833 @@
+
+    - ``"pass"``: Status reports e.g. after a successfully completed run.
+
+    - ``"comment"``: Comments regarding certain properties of the tested operation(s).
+      These does not require any action to change the operation but is used to raise awareness
+      for behaviour that might be unexpected or differing from common operations.
+
+    - ``"hint"``: Remarks similar to warnings that indicate concrete hints to change the checked
+      operation(s). The recommended changes are expected to improve the code quality, performance
+      or consistency with other operations.
+
+    - ``"error"``: Problems with the checked operation(s) that require changes. These problems
+      might be in the core of the ``Operation``, preventing instantiation, or in a specific
+      method or property that is rendered unusable or inconsistent by the problem.
+    """
+
+    def __init__(self, verbosity="pass", max_num_params=10, print_color=True, tol=1e-5):
+        # pylint: disable=too-many-instance-attributes
+        self._verbosity = {
+            key for key, val in verbosity_levels.items() if val <= verbosity_levels[verbosity]
+        }
+        self.max_num_params = max_num_params
+        self.print_color = print_color
+        self.tol = tol
+        self.results = self.output = self.tmp = self.seed = None
+
+    def __call__(self, op, parameters=None, wires=None, seed=None):
+        r"""Call the OperationChecker on one or multiple operations.
+
+        Args:
+            op (type or .operation.Operation): Operation(s) to check, either provided as class
+                or as instance. Allowed to be a ``Sequence`` of classes or instances instead,
+                in which case the function iterates over all objects.
+            parameters (Sequence[int or float]): Parameters with which the operation(s) is/are expected
+                to work. If ``op`` contains multiple types and ``parameters`` only contains one
+                set of parameters, they are broadcast to all operations.
+                Ignored for those objects in ``op`` that are not types but a class instance.
+                If not provided, random parameters are used.
+            wires (.wires.Wires): Wires with which the operation(s) is/are expected to work.
+                If ``op`` contains multiple types and ``wires`` only contains one wires object,
+                it is broadcast to all operations.
+                Ignored for those objects in ``op`` that are not types but a class instance.
+                If not provided, consecutive integer-named wires are used.
+            seed (int): Seed for random generation of parameters.
+
+        Returns:
+            dict: The result status for each checked operation, corresponding to the four levels
+            ``"error"``, ``"hint"``, ``"comment"`` and ``"pass"``.
+            dict: The text printed to the terminal for each checked operation.
```

Comment:
this seems to be repeated?

Reply:
Ah, no, it actually returns 2 `dict`s! The first is just the result _status_, whereas the second contains all text messages that are also printed to the terminal.